executor/services/workspace_archive_restore.py

```
@@ -0,0 +1,229 @@
+
+def _extract_members(
+    archive: tarfile.TarFile,
+    members: list[tarfile.TarInfo],
+    path: Path,
+) -> None:
+    if not members:
+        return
+    try:
+        archive.extractall(path=str(path), members=members, filter="data")
+    except TypeError:
+        archive.extractall(path=str(path), members=members)
```

Ruff (0.15.17)

[error] Uses of `tarfile.extractall()`

(S202)